lib/main.js

```diff
@@ -385,54 +392,192 @@ function createPipelinedRequest(cb){
+                switch (true) {
+                    case contentType.includes('application/json'):
+                        console.log('Parsing JSON response');
+                        if (responseBody) {
+                            parsedResponse = JSON.parse(responseBody);
+                        } else {
+                            console.log('Empty JSON response body');
+                            parsedResponse = {};
+                        }
+
+                        if (anonymizationEnabled) {
+                            parsedResponse = anonymizer.anonymizeObject(parsedResponse);
+                            console.log('Anonymized JSON(Response):', parsedResponse);
+                        }
+                        break;
+
+                    case contentType.includes('text/html'):
+                        console.log('Parsing HTML response');
+                        if (responseBody) {
+                            parsedResponse = responseBody;
+
+                            if (anonymizationEnabled) {
+                                parsedResponse = anonymizer.anonymizeHtml(responseBody);
+                                console.log('Anonymized HTML(Response):', parsedResponse);
+                            }
+                        } else {
+                            console.log('Empty HTML response body');
+                            parsedResponse = '';
+                        }
+                        break;
+
+                    case contentType.includes('application/x-www-form-urlencoded'):
+                        console.log('Parsing URL-encoded response');
+                        if (responseBody) {
+                            parsedResponse = anonymizer.anonymizeUrlEncoded(responseBody);
+                            console.log('Anonymized URL-encoded(Response):', parsedResponse);
+                        } else {
+                            console.log('Empty URL-encoded response body');
+                            parsedResponse = '';
+                        }
+                        break;
+
+                    default:
+                        console.log('Non-JSON response, using raw body');
+                        parsedResponse = responseBody;
+
+                        if (anonymizationEnabled) {
+                            parsedResponse = anonymizer.anonymizeText(parsedResponse);
+                            console.log('Anonymized Text(Response):', parsedResponse);
+                        }
+                        break;
+                    }
+                } catch (err) {
+                    console.error('Failed to parse response body as JSON:', err);
+                    parsedResponse = responseBody;
+                }
+    
+                if (cb) {
+                    console.log('Invoking callback with response and parsedResponse');
+                   
+                    if (cb.length === 2) {
```

Comment:
This code would not be "backwards compatible" with our main lib. See my comment at L474